src/gc/transaction_level_gc_manager.cpp

```diff
@@ -134,29 +139,31 @@ int TransactionLevelGCManager::Unlink(const int &thread_id,
     }
   }  // end for
 
-  auto safe_max_cid =
-      concurrency::TransactionManagerFactory::GetInstance().GetNextCommitId();
-  for (auto &item : garbages) {
-    reclaim_maps_[thread_id].insert(std::make_pair(safe_max_cid, item));
+  // once the current epoch id is expired, then we know all the transactions
+  // that are active at this time point will be committed/aborted.
+  // at that time point, it is safe to recycle the version.
+  eid_t safe_expired_eid = concurrency::EpochManagerFactory::GetInstance().GetCurrentEpochId();
+
+  for(auto& item : garbages){
+      reclaim_maps_[thread_id].insert(std::make_pair(safe_expired_eid, item));
```

Comment:
We should not use `std::unordered_multimap`. Our design is to sort the garbage versions according to epoch id. Versions with the oldest epoch id can be reclaimed first. Please check the code in `Reclaim()` (https://github.com/yingjunwu/peloton/blob/f5ec56f1c7f6e95f6e19a2409857f05e7ed6bb61/src/gc/transaction_level_gc_manager.cpp#L155). I have comments there.